File: CarePackage/Software/DotNet6DesktopRuntime.cs
namespace CarePackage.Software;

public class DotNet6DesktopRuntime : BaseSoftware
{
    public override string           Key            { get; set; } = "dotnet-6-desktop-runtime";
    public override string           Name           { get; set; } = ".NET 6.0 Desktop Runtime";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Runtime;
    public override string           DownloadName   { get; set; } = "windowsdesktop-runtime-6.0-win-x64.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.dotnet;
    public override string           Homepage       { get; set; } = "https://dot.net";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://download.visualstudio.microsoft.com/download/pr/f6b6c5dc-e02d-4738-9559-296e938dabcb/b66d365729359df8e8ea131197715076/windowsdesktop-runtime-6.0.36-win-x64.exe");
}

File: src/lib/software/dotnet_6_desktop_runtime.py
from src.lib.software import BaseSoftware, SoftwareCategory

class DotNet6DesktopRuntime(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'dotnet-6-desktop-runtime'
        self.name = '.NET 6.0 Desktop Runtime'
        self.category = SoftwareCategory.DotNet
        self.download_name = 'windowsdesktop-runtime-6.0-win-x64.exe'
        self.is_archive = False
        self.should_cache_url = False
        self.requires_admin = False
        self.icon = 'dotnet.png'
        self.homepage = 'https://dot.net'

    def resolve_download_url(self):
        self.url_resolved.emit('https://download.visualstudio.microsoft.com/download/pr/f6b6c5dc-e02d-4738-9559-296e938dabcb/b66d365729359df8e8ea131197715076/windowsdesktop-runtime-6.0.36-win-x64.exe')
